packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -119,14 +126,25 @@ function transform(root: IRElement, codeGen: CodeGen, state: State): t.Expressio
         return codeGen.genText(typeof value === 'string' ? value : bindExpression(value, text));
     }
 
+    function transformComment(comment: IRComment): t.Expression {
+        const { parts, dynamicIndexes } = comment;
+        const boundParts = parts.map((part) =>
+            typeof part === 'string' ? part : bindExpression(part, comment)
+        );
+
+        return codeGen.genComment(boundParts, dynamicIndexes);
+    }
+
     function transformChildren(children: IRNode[]): t.Expression {
         const res = children.reduce<t.Expression[]>((acc, child) => {
             let expr;
 
             if (isElement(child)) {
                 expr = isTemplate(child) ? transformTemplate(child) : transformElement(child);
-            } else {
+            } else if (isTextNode(child)) {
                 expr = transformText(child);
+            } else if (isCommentNode(child)) {
```

Comment:
This should be gated behind a flag. In 99% of the cases, comments in the HTML template are only used to add information for other developers. For the remaining 1%, the comment conveys some runtime semantic (eg. MSO comments). 

Using an opt-in mechanism is preferable here to avoid rendering comments in the DOM (and slowing down performance) when not needed.

Reply:
agree, though this during compilation so we should probably add a compiler option. taking note for the RFC.